desktop/src/main/java/org/destinationsol/desktop/SolDesktop.java

```diff
@@ -11,55 +26,96 @@
+ */
+public final class SolDesktop {
 
-public class SolDesktop {
     private static Logger logger = LoggerFactory.getLogger(SolDesktop.class);
 
+    /**
+     * Specifies the commandline option to pass to the application for it to generate no crash reports. Coincidentally,
+     * it is the only commandline option detected by the application.
+     */
     private static final String NO_CRASH_REPORT = "-noCrashReport";
 
-    public static void main(String[] argv) {
-        List<String> commandLineArgs = Arrays.asList(argv);
+    /**
+     * This class is basically only a holder for the Java's {@code main(String[])} method, thus needs not to be
+     * instantiated.
+     */
+    private SolDesktop() {
+    }
 
+    public static void main(String[] argv) {
         LwjglApplicationConfiguration applicationConfig = new LwjglApplicationConfiguration();
-        boolean devBuild = java.nio.file.Files.exists(Paths.get("devBuild"));
-        if (devBuild) {
-            DebugOptions.DEV_ROOT_PATH = "engine/src/main/resources/"; // Lets the game run from source without a tweaked working directory
-            applicationConfig.vSyncEnabled = false; // Setting to false disables vertical sync
-            applicationConfig.foregroundFPS = 100; // Use 0 to disable foreground fps throttling
-            applicationConfig.backgroundFPS = 10; // Use 0 to disable background fps throttling
-        }
+        //TODO: Is checking for a presence of the file really the way we want to determine if it is a debug build?
+        handleDevBuild(applicationConfig);
         MyReader reader = new MyReader();
         DebugOptions.read(reader);
 
-        if (DebugOptions.EMULATE_MOBILE) {
-            applicationConfig.width = 640;
-            applicationConfig.height = 480;
-            applicationConfig.fullscreen = false;
-        } else {
-            GameOptions d = new GameOptions(false, reader);
-            applicationConfig.width = d.x;
-            applicationConfig.height = d.y;
-            applicationConfig.fullscreen = d.fullscreen;
-        }
+        // Set screen width, height...
+        setScreenDimensions(applicationConfig, reader);
 
+        // Set the application's title, icon...
         applicationConfig.title = "Destination Sol";
         if (DebugOptions.DEV_ROOT_PATH == null) {
             applicationConfig.addIcon("src/main/resources/icon.png", Files.FileType.Internal);
         } else {
             applicationConfig.addIcon(DebugOptions.DEV_ROOT_PATH + "/icon.png", Files.FileType.Absolute);
         }
 
-        if (!commandLineArgs.contains(NO_CRASH_REPORT)) {
+        handleCrashReporting(argv);
+
+        // Everything is set up correctly, launch the application
+        new LwjglApplication(new SolApplication(), applicationConfig);
+    }
+
+    /**
+     * When on dev build, use specific settings.
+     *
+     * Whether a build is a dev build is found out by checking of a file "devBuild" in the root directory of DestSol.
```

Comment:
This may be getting a little conversational again, for javadoc (great for tutorials and such tho)

Reply:
Improved the first sentence to give more overall look to the method, the rest _is_ basically tutorial (or, I don't see happening anything more tutorial-y anytime soon, so if anyone tries to learn DS codebase, javadocs should provide help)